### DISCUSSION TITLE
Migrated repo to Git LFS, clones much slower now

### DISCUSSION THREAD

**Discussion by robertjdac** [git-lfs/git-lfs#4740] [Q&A] [answered]
Our repo was migrated from svn to git a year ago, hosting on our company's Gitlab server. We were new to git so weren't aware of the issues with adding binary files, of which our svn repo contained a lot (which frequently changed).

This past week I migrated the repo to use git lfs following this guide: https://github.com/git-lfs/git-lfs/wiki/Tutorial#migrating-existing-repository-data-to-lfs

It worked great for me and most developers on my team. Unfortunately, two of the developers are having an issue with their clone operation takes hours (versus minutes for the rest of us). This is worse than pre-migration to LFS.

The big difference I can see from their clone is this:

For those without issue, our clone's last step is "Filtering content".

However, for the two with this issue, their console prints out "Downloading ..." every single binary file. This step takes forever as it iterates over every single binary file, downloading them one by one.

I don't see anything that stands out in their git config that could cause this. Just wondering if anyone has encountered an issue with clones taking forever after migration to git lfs. I tried googling but couldn't find anyone with this issue.

**Comment by bk2204** [comment, marked as answer]
Hey,

Git LFS is based on smudge and clean filters.  There are versions of these which process files incrementally and those that process using the filter process.  The former is unsurprisingly much slower and is the cause of the "Downloading" message and the latter is much more efficient and causes the "Filtering content" messages.

For the users seeing slow behavior, you'll want to have them run `git lfs env`.  They should be running at least (I believe) Git 2.15.0 and you should see the following lines in the output:

```
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

If you don't see the first line, that's probably the issue, which you can resolve with `git lfs install`.  If they're using an older version of Git, they should upgrade.

**Reply by robertjdac** [reply]
Hello, thanks for the reply. Does everyone on the team need to run `git lfs install`? I ran it once when migrating the repo, but assumed it wasn't necessary for everyone. I'm guessing they have to run that before they clone?

**Reply by bk2204** [reply]
Everyone needs to have run it once on their system or have configured the smudge and clean filters it sets up manually (e.g., I have the settings in my dotfiles).  Ideally they will have run it before they clone, yes.

**Reply by robertjdac** [reply]
For the two impacted developers, I had them run git lfs install and then git lfs env. Same issue with super slow clone afterwards. These are the results:

$ git lfs env
git-lfs/2.12.0 (GitHub; windows amd64; go 1.14.7; git dad7b9e6)
git version 2.29.2.windows.1

Error reading git config: error running C:\Program Files\Git\mingw64\libexec\git-core\git.exe 'rev-parse' '--is-bare-repository': 'fatal: not a git repository (or any of the parent directories): .git' 'exit status 128'
LocalWorkingDir=
LocalGitDir=
LocalGitStorageDir=
LocalMediaDir=lfs\objects
LocalReferenceDirs=
TempDir=lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_LFS_PATH=C:\Program Files\Git LFS
GIT_SSHx=C:\Program Files\Git\usr\bin\ssh.exe
git config filter.lfs.process = ""
git config filter.lfs.smudge = ""
git config filter.lfs.clean = ""


For what it's worth, me and the other developers have same version and same output from git lfs env, but we don't suffer from the super slow clone issue.

**Reply by bk2204** [reply]
Try running the command in a repository, since older versions of `git lfs env` can fail in some cases when not run in a repository.

**Reply by robertjdac** [reply]
$ git lfs env
git-lfs/2.12.0 (GitHub; windows amd64; go 1.14.7; git dad7b9e6)
git version 2.29.2.windows.1

Endpoint=https://gitlab.us.redacted/bh-mod-sw/cockpit-sw.git/info/lfs (auth=basic)
LocalWorkingDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw
LocalGitDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw\.git
LocalGitStorageDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw\.git
LocalMediaDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw\.git\lfs\objects
LocalReferenceDirs=
TempDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw\.git\lfs\tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=D:\BH_Mod\fms_BHMOD-412\cockpit-sw\.git\lfs
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic,lfs-standalone-file
UploadTransfers=basic,lfs-standalone-file
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_LFS_PATH=C:\Program Files\Git LFS
GIT_SSHx=C:\Program Files\Git\usr\bin\ssh.exe
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"

**Reply by robertjdac** [reply]
There are a few parameters that differ between my config and the impacted developer (mine works, his doesn't). Here's the differences:

Endpoint=https://gitlab.us.redacted/bh-mod-sw/cockpit-sw.git/info/lfs (auth=none)
  SSH=git@gitlab.us.redacted.com:bh-mod-sw/cockpit-sw.git 

AccessDownload=none
AccessUpload=none 

So I have auth=none and AccessDownload and AccessUpload = none
